display/ssd1306.h

```diff
@@ -225,15 +229,37 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	if (eof_) {
 	  // STDOUT << "EOF " << frame_count_ << "\n";
-	  screen_ = SCREEN_PLI;
-	  if (frame_count_ == 1) return 5000;
-	  return FillFrameBuffer();
+    if (!SaberBase::IsOn()) {
+	   screen_ = SCREEN_PLI;
+	    if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+	    return FillFrameBuffer();
+    }
 	}
 	frame_count_++;
-	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
-	}
+  if (SaberBase::IsOn()) {
+    if (looped_frames_ == 1 && frame_count_ == 1) {
+      return font_config.ProffieOSImageDuration;
+    } else if (looped_frames_ == 1 && frame_count_ > 1) {
+      if (!SaberBase::Lockup()) {
```

Comment:
I think we should invert this if statement. (and swap the two code blocks below)
Same for the one on the next line.

Reply:
done